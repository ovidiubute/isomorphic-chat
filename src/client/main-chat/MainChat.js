import React from "react";

class MainChat extends React.Component {
  render() {
    return <p>Hello from the backend!</p>;
  }
}

export default MainChat;
