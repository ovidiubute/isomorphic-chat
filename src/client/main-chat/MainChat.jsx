/* eslint-disable react/prop-types */
import React from "react";

class MainChat extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      messages: [],
      username: null
    };
  }

  componentWillMount() {
    this.setState((prevState, props) => ({
      username: `anon-${props.socket.id}`
    }));

    if (this.props.socket) {
      this.props.socket.on("message", this.receiveMessage);
    }
  }

  onKeyUp = e => {
    if (e.keyCode === 13) {
      this.props.socket.emit("message", e.target.value);
    }
  };

  receiveMessage = message => {
    this.setState(prevState => ({
      messages: prevState.messages.concat(message)
    }));
  };

  render() {
    return (
      <div>
        <input type="text" defaultValue="Hey" onKeyUp={this.onKeyUp} />
        <ul>
          {this.state.messages.map(m => <li>{m}</li>)}
        </ul>
      </div>
    );
  }
}

export default MainChat;
