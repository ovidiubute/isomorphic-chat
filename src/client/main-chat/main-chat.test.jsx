import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import MainChat from "./main-chat";

describe("MainChat", () => {
  it("should render in empty state", () => {
    const r = ReactShallowRenderer.createRenderer();
    r.render(<MainChat />);

    const result = r.getRenderOutput();
    expect(result.type).toBe("div");
    expect(result.props.children).toHaveLength(2);
    expect(result.props.children[0].type).toBe("ul");
    expect(result.props.children[1].type).toBe("input");
  });
});
