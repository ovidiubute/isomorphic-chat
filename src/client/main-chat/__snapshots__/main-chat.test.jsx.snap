// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainChat should render with messages 1`] = `
<div
  style={
    Object {
      "height": "80vh",
      "margin": "auto",
      "maxHeight": "80vh",
      "width": 600,
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "white",
        "border": "1px solid #d3d3d3",
        "borderRadius": 8,
        "color": "rgb(89, 68, 68)",
        "height": "inherit",
        "maxHeight": "inherit",
        "overflow": "scroll",
      }
    }
  >
    <ul
      style={
        Object {
          "listStyleType": "none",
          "margin": 0,
          "padding": 8,
        }
      }
    >
      <li>
        <p
          style={
            Object {
              "borderRadius": 8,
              "boxShadow": "2px 2px 2px 1px rgba(0, 0, 0, 0.2)",
              "display": "inline-block",
              "margin": "0 0 12px 0",
              "overflowWrap": "break-word",
              "padding": 8,
              "userSelect": "text",
              "verticalAlign": "baseline",
            }
          }
        >
          <span
            style={
              Object {
                "color": "peru",
                "fontSize": "x-small",
                "verticalAlign": "middle",
              }
            }
          >
            daniel: 
          </span>
          <span>
            Hi
          </span>
        </p>
      </li>
      <li>
        <p
          style={
            Object {
              "borderRadius": 8,
              "boxShadow": "2px 2px 2px 1px rgba(0, 0, 0, 0.2)",
              "display": "inline-block",
              "margin": "0 0 12px 0",
              "overflowWrap": "break-word",
              "padding": 8,
              "userSelect": "text",
              "verticalAlign": "baseline",
            }
          }
        >
          <span
            style={
              Object {
                "color": "peru",
                "fontSize": "x-small",
                "verticalAlign": "middle",
              }
            }
          >
            josh: 
          </span>
          <span>
            Bye
          </span>
        </p>
      </li>
    </ul>
  </div>
  <input
    onChange={[Function]}
    onKeyUp={[Function]}
    style={
      Object {
        "border": "1px solid #d3d3d3",
        "float": "left",
        "fontSize": 24,
        "marginTop": 4,
        "outline": "none",
        "padding": 4,
        "width": 500,
      }
    }
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#abe37f",
        "border": 0,
        "color": "white",
        "fontSize": 16,
        "height": 38,
        "marginTop": 4,
        "outline": "none",
        "width": 100,
      }
    }
    type="button"
  >
    SEND
  </button>
</div>
`;
